algorithms/pigeonholeSort.py

```diff
@@ -1,20 +1,34 @@
+# for pigeonholes, the indexes represent:
+# the value from the array - the minimum value,
+# and the item is the number of times that value is in the array
+# we initialize the counts to 0
+def pigeonholes(size):
+  return [0]*size
+
+def addTo(array, value):
+  array[value] += 1
+
+def subtractFrom(array, value):
+  array[value] -= 1
+
 def pigeonholeSort(array, *args):
-  minV = min(array)
-  maxV = max(array)
-  A = array.copy()
-  size = maxV - minV + 1
-  C = [0]*(size)
-  i = 0
-  for x in array:
-    yield A, i, -1, -1, -1
-    C[x - minV] += 1
-    A[i] = A[i] / (C[x-minV] + 1)
```

Comment:
This calculation was doing something with the blue line, but it didn't make any sense with the animation so I just removed it.